sphinx/ext/autodoc/__init__.py

```diff
@@ -1468,17 +1464,23 @@ def format_args(self, **kwargs: Any) -> str:
             kwargs.setdefault('show_annotation', False)
 
         unwrapped = inspect.unwrap(self.object)
-        if ((inspect.isbuiltin(unwrapped) or inspect.ismethoddescriptor(unwrapped)) and
-                not inspect.is_cython_function_or_method(unwrapped)):
-            # can never get arguments of a C function or method
-            return None
-        if inspect.isstaticmethod(unwrapped, cls=self.parent, name=self.object_name):
-            self.env.app.emit('autodoc-before-process-signature', unwrapped, False)
-            sig = inspect.signature(unwrapped, bound_method=False)
-        else:
-            self.env.app.emit('autodoc-before-process-signature', unwrapped, True)
-            sig = inspect.signature(unwrapped, bound_method=True)
-        args = stringify_signature(sig, **kwargs)
+        try:
+            if self.object == object.__init__ and self.parent != object:
+                # Classes not having own __init__() method are shown as no arguments.
+                #
+                # Note: The signature of object.__init__() is (self, /, *args, **kwargs).
+                #       But it makes users confused.
+                args = '()'
+            else:
+                if inspect.isstaticmethod(unwrapped, cls=self.parent, name=self.object_name):
+                    self.env.app.emit('autodoc-before-process-signature', unwrapped, False)
+                    sig = inspect.signature(unwrapped, bound_method=False)
+                else:
+                    self.env.app.emit('autodoc-before-process-signature', unwrapped, True)
+                    sig = inspect.signature(unwrapped, bound_method=True)
+                args = stringify_signature(sig, **kwargs)
+        except (TypeError, ValueError):
```

Comment:
We should not handle `TypeError` simply as discussed. I'll handle it on another PR.